xbmc/Application.cpp

```diff
@@ -2252,6 +2252,19 @@ void CApplication::OnApplicationMessage(ThreadMessage* pMsg)
   }
   break;
 
+  case TMSG_SET_PLAYER_ITEM:
```

Comment:
it seems really odd that some component (in this case retroplayer) sends some thread message to the application so that it resets its internal state. Can't this be done in some other way, for example by extending the `IPlayerCallback` interface? This has been the cause for a lot of bugs regarding state inconsistency between the app and the player (in json-rpc for example).

I took a look into the python addon and you seem to follow some polling approach in the service. If callbacks are used instead you could just react to the `OnSomething` event

Reply:
I wonder if this is needed at all.
Retroplayer calls `OnPlaybackStarted` just as VideoPlayer or any other player.
CApplications `OnPlayBackStarted` handler sets `m_itemCurrentFile` already.

Reply:
Also `CServiceBroker::GetGUI()->GetInfoManager().SetCurrentItem(*m_itemCurrentFile);` is done in in 
GUI_MSG_PLAYBACK_STARTED of CApplication (and this message is sent at the end of the OnPlaybackStarted callback handler, so this code ultimately runs already as well).

My take: If it does not work without this new message, you should probably find out the root cause and find a better solution to your problem.

Reply:
My guess is that maybe during playback the item changes without restarting the player (some video game level or something similar?) and you get to a point where there is state inconsistency between the item the app holds and the item the player holds. That's why my suggestion was to extend the player callback intertface with some other event that represents what happens here if this is really needed. Basically following the same approach that already takes place `OnPlaybackStarted`

Reply:
Not sure how the item would change without a call to `OpenFile`.

Reply:
The item tags change without opening an other file, every time the game saves a caption is generated and it might change depending your progress

Reply:
It's worth thinking that maybe we took the wrong approach. The need is to pass the caption to Python. We use the player API and pass game info tags of the currently-playing item. The advantage of going through the player is that we don't need to worry about modifying the Python API. But maybe, to avoid the loop mentioned in the [comment added to CRetroPlayer](https://github.com/xbmc/xbmc/pull/21286/commits/b1529dc1e58f011e96b5f3b81bbe4462d887dd32#diff-9ce90c3637f8cbc50ea1ae92ba5eb1eef4c601d472eb0946430086b1067e29bbR587), we modify the player API?